fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/RecordMetaData.java

```diff
@@ -475,6 +475,27 @@ public static KeyExpression commonPrimaryKey(@Nonnull Collection<RecordType> rec
         return common;
     }
 
+    /**
+     * Calculate and return the length common to all the PKs of the given types. The common PK length is calculated by
+     * ensuring ALL PKs for the given types have the same length (number of fields). If there is no common length, 0 is
+     * returned.
+     * @param recordTypes the (non-null) collection of record types to calculate the common PK length for
+     * @return the common length for all given types, 0 if not found
+     */
+    public static int commonPrimaryKeyLength(@Nonnull Collection<RecordType> recordTypes) {
+        int common = 0;
+        boolean first = true;
+        for (RecordType recordType : recordTypes) {
+            if (first) {
+                common = recordType.getPrimaryKey().getColumnSize();
+                first = false;
+            } else if (common != recordType.getPrimaryKey().getColumnSize()) {
+                return 0;
```

Comment:
Instead of returning 0 when a match isn't found, this feels like it should either return -1 or the whole method should return an `OptionalInt`. I think a zero-length primary key might actually be possible (with an `EMPTY` key expression), or at least, it seems safer to act like it could here, just in case.

Reply:
Done.